lib/shared/store.js

```diff
@@ -414,24 +426,37 @@ const storeReducer = (state = DEFAULT_STATE, action) => {
         })
       }
 
-      const selectedDrive = findDrive(state, state.getIn([ 'selection', 'drive' ]))
+      const selectedDevices = state.getIn([ 'selection', 'drives' ])
 
-      return _.attempt(() => {
-        if (selectedDrive && !_.every([
-          constraints.isDriveValid(selectedDrive.toJS(), action.data),
-          constraints.isDriveSizeRecommended(selectedDrive.toJS(), action.data)
-        ])) {
-          return storeReducer(state, {
-            type: ACTIONS.REMOVE_DRIVE
+      return _.reduce(selectedDevices.toJS(), (accState, driveDevice) => {
+        const drive = findDrive(state, driveDevice)
+        if (drive && !constraints.isDriveValid(drive.toJS(), action.data)) {
+          return storeReducer(accState, {
+            type: ACTIONS.REMOVE_DRIVE,
+            data: driveDevice
           })
         }
 
-        return state
-      }).setIn([ 'selection', 'image' ], Immutable.fromJS(action.data))
+        return accState
+      }, state).setIn([ 'selection', 'image' ], Immutable.fromJS(action.data))
```

Comment:
So it looks like this e.g. removes every selected drive that is now smaller than the newly-selected image? There should definitely be unit-tests for this ;-)